Tests/ConcurrencyHelpersTests/ConcurrencyHelpersTests.swift

```diff
@@ -28,4 +28,38 @@ final class ConcurrencyHelpersTests: XCTestCase {
 
         lock.withLockVoid { _ = 4 }
     }
+
+    struct Data {
+        @Protected public var valueA: Int = 1
+        @Protected public var valueB: Int? = nil
+
+        public mutating func setA(_ a: Int) {
+            _valueA.write {
+                $0 = a
+            }
+        }
+    }
+
+    func testProceted() {
```

Comment:
I'd test that such properties can be safely accessed from multiple threads.

Reply:
I've added basic tests only for now, kind of sanity and accessibility checks. Full tests suit to be added later